Asterix/optics/propagation_functions.py

```diff
@@ -585,55 +586,98 @@ def prop_fpm_regional_sampling(pup,
 
     AUTHOR: R. Galicher (in IDL)
             ILa (to Python)
+            J. Mazoyer : small modifs to adapt to Asterix Dec 22
 
     Parameters
     ----------
-    pup : 2D array
+    pup : 2D array of size [self.dim_overpad_pupil, self.dim_overpad_pupil]
         Input mage array containing the wavefront at the entrance pupil of the optical system.
     fpm : 2D array
         Complex electric field in the focal plane of the focal-plane mask.
-    nbres : 1D array or list
-        List of the number of resolution elements in the total image plane for all propagation layers.
+    real_dim_input : int or None, default None
+        Diameter of the support in pup (can differ from pup.shape). If None, real_dim_input = pup.shape
+        If the pupil fills its array out until the edges, then real_dim_input = pup.shape; if the pupil
+        is padded, then real_dim_input is the size of the pupil in pixels.
+    nbres : list
+        List of the number of resolution elements across the total image plane for all propagation layers.
+        As a general rule, it is probably safest to choose these numbers such that there is not sampling
+        shift right in the middle of the DH. This would ensure that the frequencies inside the DH are all
+        calculated with the same resolution in the EFC matrix. So if we correct the DH between two radii
+        IWA and OWA (in lambda/D), nbres should not have any elements between 2IWA and 2OWA.
     shift : tuple, default (0, 0)
-        Shift of FPM with respect to optical axis in units of lambda/D. This is done by introducing a tip/tilt on the
-        input wavefront in the pupil that is subsequently taken out in the Lyot plane after each propagation layer.
-    samp_outer : float
-        Sampling in the outermost layer of propagations.
+        Shift of FPM with respect to optical axis in units of lambda/D.
     filter_order : int
         Order of the Butterworth filter.
     alpha : float
         Scale factor for the filter size. The larger this number, the smaller the filter size with respect to the
         input array.
+    dir_save_all_planes : string, default None
+        Directory to save all planes into fits files if save_all_planes_to_fits=True.
```

Comment:
I think this is not quite correct, the variable `save_all_planes_to_fits` doesn't show up anywhere in here.

Reply:
Oops I copied and past another docstring that was apparently outdated. Changed both